### PR TITLE
Copter: Enumerate A and B points of ZigZag

### PR DESCRIPTION
**Pull request by muramura**
I reflected the comment of #13648 
I decided to define points A and B by enum.

### CODE REVIEW

File: ArduCopter/mode.h [hunk 1]
```diff
@@ -1390,6 +1390,11 @@ class ModeZigZag : public Mode {
     // Inherit constructor
     using Mode::Mode;
 
+    enum class position : uint8_t {
```
Comment (peterbarker) [inline ArduCopter/mode.h]: Capital-P for position, please.

Or, as stated below, capital-D-for-Destination.

File: ArduCopter/mode.h [hunk 2]
```diff
@@ -1398,8 +1403,8 @@ class ModeZigZag : public Mode {
     bool allows_arming(bool from_gcs) const override { return true; }
     bool is_autopilot() const override { return true; }
 
-    // save current position as A (dest_num = 0) or B (dest_num = 1).  If both A and B have been saved move to the one specified
-    void save_or_move_to_destination(uint8_t dest_num);
+    // save current position as A (dest_num = position::A) or B (dest_num = position::B).  If both A and B have been saved move to the one specified
+    void save_or_move_to_destination(position dest_num);
```
Comment (peterbarker) [inline ArduCopter/mode.h]: This is no longer `dest_num`.

Also, `Destination destination` might be better as we use `Position` for locations-relative-to-origin.

**Comment by peterbarker** [comment]
@muramura this branch is now conflicting; are you still interested in pushing this in?

**Comment by khancyr** [comment]
merged as part of https://github.com/ArduPilot/ardupilot/pull/14020

**Pull request closed without merging**